From 24d7f92d03e06442d3eae45d0c5b945e484a39da Mon Sep 17 00:00:00 2001
From: xuxiang <xuxiang@eswincomputing.com>
Date: Mon, 22 Apr 2024 16:27:39 +0800
Subject: [PATCH 003/109] WIN2030-15279 : fix : add production test bug

Changelogs:
1. Delete part of the debugging process
2. Remove unnecessary printing

Change-Id: Idc5a546e57be4e809ef73d97dcb404304ba3df86
---
 Core/Src/hf_power_process.c      |  1 -
 Core/Src/hf_protocol_process.c   | 12 +++++-------
 Core/Src/protocol_lib/protocol.c |  4 ++--
 3 files changed, 7 insertions(+), 10 deletions(-)

diff --git a/Core/Src/hf_power_process.c b/Core/Src/hf_power_process.c
index f1ff30f..0903dab 100644
--- a/Core/Src/hf_power_process.c
+++ b/Core/Src/hf_power_process.c
@@ -35,7 +35,6 @@ void hf_power_task(void *parameter)
 	HAL_StatusTypeDef status = HAL_OK;
 	power_state_t power_state = IDLE_STATE;
 	GPIO_PinState pin_state = GPIO_PIN_RESET;
-	printf("\r\n%s %d %s %s\r\n", __func__, __LINE__, __DATE__, __TIME__);
 	#ifdef AUTO_BOOT
 	power_state = ATX_PS_ON_STATE;
 	som_power_state = SOM_POWER_ON;
diff --git a/Core/Src/hf_protocol_process.c b/Core/Src/hf_protocol_process.c
index ca2d8c6..cbe832c 100644
--- a/Core/Src/hf_protocol_process.c
+++ b/Core/Src/hf_protocol_process.c
@@ -25,7 +25,6 @@ void es_send_req(b_frame_class_t *pframe, uint8_t req_cmd, int8_t *frame_data, u
 {
 	uint8_t buf[MAX_FRAME_LEN] = {0};
 	uint8_t b_len, xor = 0;
-	printf("%s\n", frame_data);
 	for (int i = 0; i < len; i++) {
 		xor ^= frame_data[i];
 	}
@@ -45,9 +44,6 @@ void es_send_req(b_frame_class_t *pframe, uint8_t req_cmd, int8_t *frame_data, u
 
 	memcpy(buf + b_len, pframe->frame_info.end, pframe->frame_info.end_len);
 	b_len += pframe->frame_info.end_len;
-	printf("len = %d \n", len);
-	printf("xor = %d \n", xor);
-	return 0;
 
 	HAL_UART_Transmit(&huart3, buf, b_len, HAL_MAX_DELAY); // transmit the full sentence again
 }
@@ -133,11 +129,12 @@ int32_t es_set_eth(struct ip_t *ip, struct netmask_t *netmask, struct getway_t *
 int32_t es_set_rtc_date(struct rtc_date_t *sdate)
 {
 	RTC_DateTypeDef sDate = {0};
-	sDate.Year = sdate->Year;
+	sDate.Year = sdate->Year - 2000;
 	sDate.Month = sdate->Month;
 	sDate.Date = sdate->Date;
 	sDate.WeekDay = sdate->WeekDay;
 
+	// printf("yy/mm/dd  %02d/%02d/%02d\r\n", sDate.Year, sDate.Month, sDate.Date);
 	if (HAL_RTC_SetDate(&hrtc, &sDate, RTC_FORMAT_BCD) != HAL_OK)
 		return HAL_ERROR;
 	return HAL_OK;
@@ -161,7 +158,7 @@ int32_t es_get_rtc_date(struct rtc_date_t *sdate)
 	RTC_DateTypeDef GetData;
 	if (HAL_RTC_GetDate(&hrtc, &GetData, RTC_FORMAT_BIN) != HAL_OK)
 		return HAL_ERROR;
-	printf("yy/mm/dd  %02d/%02d/%02d\r\n", 2000 + GetData.Year, GetData.Month, GetData.Date);
+	// printf("yy/mm/dd  %02d/%02d/%02d\r\n", 2000 + GetData.Year, GetData.Month, GetData.Date);
 	sdate->Year = 2000 + GetData.Year;
 	sdate->Month = GetData.Month;
 	sdate->Date = GetData.Date;
@@ -225,7 +222,7 @@ int32_t es_get_rtc_time(struct rtc_time_t *stime)
 	RTC_TimeTypeDef GetTime;
 	if (HAL_RTC_GetTime(&hrtc, &GetTime, RTC_FORMAT_BIN) != HAL_OK)
 		return HAL_ERROR;
-	printf(" hh:mm:ss %02d:%02d:%02d\r\n", GetTime.Hours, GetTime.Minutes, GetTime.Seconds);
+	// printf(" hh:mm:ss %02d:%02d:%02d\r\n", GetTime.Hours, GetTime.Minutes, GetTime.Seconds);
 	stime->Hours = GetTime.Hours;
 	stime->Minutes = GetTime.Minutes;
 	stime->Seconds = GetTime.Seconds;
@@ -334,6 +331,7 @@ void protocol_task(void *argument)
 			if (!es_get_cmd_and_data(&frame_uart3)) {
 				protocol_status = CHECK_TAIL;
 			} else {
+				// printf("%s CHECK_CMD fail\n", __func__);
 				es_send_req(&frame_uart3, CMD_RES, req_fail, sizeof(req_fail) - 1);
 				protocol_status = CHECK_HEAD;
 			}
diff --git a/Core/Src/protocol_lib/protocol.c b/Core/Src/protocol_lib/protocol.c
index 3fe7ff7..68bc28b 100644
--- a/Core/Src/protocol_lib/protocol.c
+++ b/Core/Src/protocol_lib/protocol.c
@@ -83,7 +83,7 @@ uint8_t es_get_cmd(b_frame_class_t *pframe)
 	uint8_t cmd, ret = B_ERROR;
 	if (read_ring_buf(&pframe->_frame_ring, &cmd, 1) == 1) {
 		// printf("cmd %x\n", cmd);
-		if (CMD_EEPROM_WP <= cmd && CMD_SET_IP >= cmd) {
+		if (CMD_EEPROM_WP <= cmd && CMD_RES >= cmd) {
 			pframe->frame.cmd = cmd;
 			ret = B_SUCCESS;
 		}
@@ -112,7 +112,7 @@ uint8_t es_check_frame(b_frame_class_t *pframe)
 		// printf("buf[%d] %x\n", i, buf[i]);
 	}
 
-	// printf("%s %d\n", __func__, __LINE__);
+	// printf("%s %d current_xor %x\n", __func__, __LINE__, current_xor);
 	if (read_ring_buf(&pframe->_frame_ring, &xor, 1) != 1)
 		return B_ERROR;
 	// printf("%s %d\n", __func__, __LINE__);
-- 
2.25.1

